src/plays/typing-speed-test/components/Main.js

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState, useRef } from "react";
+import { VscRefresh } from "react-icons/vsc";
+
+// Project local imports
+import { generateWords } from "../utils";
+import Stats from "./Stats";
+import Timer from "./Timer";
+import Word from "./Word";
+import ResultModal from "./ResultModal";
+
+const Main = () => {
+  const userInputRef = useRef(null);
+  const [words, setWords] = useState("");
+  const [userInput, setUserInput] = useState("");
+  const [timer, setTimer] = useState(60);
+  const [status, setStatus] = useState("waiting");
+  const [activeWordIndex, setActiveWordIndex] = useState(0);
+  const [typedWordsArray, setTypedWordsArray] = useState([]);
+  const [incorrectWords, setIncorrectWords] = useState(0);
+  const [correctWords, setCorrectWords] = useState(0);
+  const [correctCharLength, setCorrectCharLength] = useState(0);
+  const [isTimerStart, setIsTimerStart] = useState(false);
+  const [isResultModalOpen, setIsResultModalOpen] = useState(false);
+
+  const refreshState = () => {
+    setWords(generateWords());
+    setUserInput("");
+    setStatus("waiting");
+    setIsTimerStart(false);
+    setTimer(60);
+    setActiveWordIndex(0);
+    setCorrectWords(0);
+    setIncorrectWords(0);
+    setCorrectCharLength(0);
+    setTypedWordsArray([]);
+    userInputRef.current.focus();
+  };
+
+  const handleModalClose = () => {
+    refreshState();
+    setIsResultModalOpen(false);
+  };
+
+  const checkIsWordMatch = (value) => {
+    // If it ends with space it meanse user has finished the word
+    setActiveWordIndex((index) => index + 1);
+    setUserInput("");
+
+    // To check each character and word
+    let wordToCompare = words[activeWordIndex];
+    let isWordMatch = wordToCompare === value.trim();
+
+    setTypedWordsArray([...typedWordsArray, isWordMatch]);

```

Comment:
@makdoom , Here we are adding a boolean `isWordMatch` to the `typedWordsArray`, can you explain what we are trying to achieve here?

Reply:
`typedWordsArray` contains all the words typed by the user weather it is correct or not... then based on that the `Word` component render the word if `Word` component gets `correct` prop as `true` then it will return green color word (it means correct word) and if it gets `correct` prop as `false` then it will return red color word